src/toggle/toggle.tsx

```diff
@@ -0,0 +1,157 @@
+                        {
+                            if (!selected) {
+                                setSelected((prevSelected) => {
+                                    return !prevSelected;
+                                });
+                            }
+                        }
+                        break;
+                    case "radio":
+                    case "yes":
+                    case "no":
+                        {
+                            if (!selected) {
+                                setSelected(true);
+                            }
+                        }
+                        break;
+                }
+
+                if (onChange) onChange(event);
+            }
+        },
+        []
+    );
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderIndicator = () => {
+        let toggleIconType: ToggleIconType;
+
+        switch (type) {
+            case "yes":
+                toggleIconType = "tick";
+                break;
+            case "no":
+                toggleIconType = "cross";
+                break;
+            case "checkbox":
+            case "radio":
+                toggleIconType = type;
+                break;
+        }
+
+        return (
+            <ToggleIcon
+                type={toggleIconType}
+                active={selected}
+                disabled={disabled}
+            />
+        );
+    };
+
+    // TODO: Reinstate later
+    // const renderSubLabel = () => {
+    //     if (!subLabel) {
+    //         return null;
+    //     }
+
+    //     let component: string | JSX.Element;
+
+    //     if (typeof subLabel === "string") {
+    //         component = subLabel;
+    //     } else if (typeof subLabel === "function") {
+    //         component = subLabel();
+    //     }
+
+    //     return (
+    //         <SubLabel disabled={disabled} $selected={selected}>
+    //             {component}
+    //         </SubLabel>
+    //     );
+    // };
+
+    return (
+        <Container
+            $selected={selected}
+            disabled={disabled}
+            role={type === "checkbox" ? "checkbox" : "radio"}
+            aria-checked={selected}
+            className={className}
+            $styleType={styleType}
+            $error={error}
+            $indicator={indicator}
+            // $hasSubLabel={!!subLabel}
+            id={id}
+            data-testid={testId}
+        >
+            {indicator && renderIndicator()}
+            <Input
+                ref={inputRef}
+                type={type === "checkbox" ? "checkbox" : "radio"}
+                data-testid="toggle-input"
+                disabled={disabled}
+                onChange={handleOnChange}
+                checked={selected}
+            />
+            <TextContainer>
+                <Label
```

Comment:
This is a good point. Initially i saw this as a label for a button but I get where you are coming from since it's a checkbox or radio input under the hood. And it'll be unlikely that someone will still put an explicit label for this.

I went for the generated short uuid or prop passed id approach as it feels cleaner structurally.